### PR TITLE
Add initial support for Star Emperor/Soul Reaper

### PR DESCRIPTION
**Pull request by Asheraf**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

This adds placeholders for the new jobs. if something is missing please make a comment so i can add it.

credit: partially ported from rytech16 - 3ceam[r850]

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2049)

File: src/map/battle.c [hunk 1]
```diff
@@ -5271,9 +5271,9 @@ struct Damage battle_calc_weapon_attack(struct block_list *src,struct block_list
 				case RL_B_FLICKER_ATK:
 					rskill = RL_FLICKER;
 					break;
-				case RL_GLITTERING_GREED_ATK:
-					rskill = RL_GLITTERING_GREED;
-					break;
+				// case RL_GLITTERING_GREED_ATK:
+				//	rskill = RL_GLITTERING_GREED;
+				//	break;
 				default:
```
Comment (Jedzkie) [inline src/map/battle.c]: Is this intended?

Comment (AnisotropicDefixation) [reply src/map/battle.c]: Should just remove it altogether as the skill doesn't exist.

File: db/pre-re/job_db.conf [hunk 2]
```diff
@@ -1669,3 +1669,25 @@ Summoner: {
 			 408, 410, 413, 415, 418, 420, 423, 425, 428, 430,	// 161 - 170
 			 433, 435, 438, 440, 443]							// 171 - 175
 }
+
+// Official values for Star_Emperor and Soul_Reaper still unknown, using place holders for now
+Star_Emperor:{
+	BaseExpGroup: "ExtendedClasses"
```
Comment (dastgirp) [inline db/pre-re/job_db.conf]: ExtendedClasses group does not exist in pre-renewal

File: src/map/map.h [hunk 3]
```diff
@@ -297,21 +299,22 @@ enum {
...
+	MAPID_BABY_CROSS           = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_THIEF,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_TAEKWON,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_WEDDING,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_GUNSLINGER,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_NINJA,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_XMAS,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_SUMMER,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_GANGSI,
+	//                         = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_SUMMONER,
+	MAPID_BABY_STAR_EMPEROR    = JOBL_THIRD | JOBL_BABY | JOBL_2_1 | MAPID_STAR_GLADIATOR,
```
Comment (dastgirp) [inline src/map/map.h]: Star Emperor and new jobs shouldn't be 3rd job

File: src/map/map.h [hunk 4]
```diff
@@ -329,6 +332,7 @@ enum {
 	//                          = JOBL_THIRD | JOBL_BABY | JOBL_2_2 | MAPID_SUMMER,
 	//                          = JOBL_THIRD | JOBL_BABY | JOBL_2_2 | MAPID_GANGSI,
 	//                          = JOBL_THIRD | JOBL_BABY | JOBL_2_2 | MAPID_SUMMONER,
+	MAPID_BABY_SOUL_REAPER      = JOBL_THIRD | JOBL_BABY | JOBL_2_2 | MAPID_SOUL_LINKER,
```
Comment (dastgirp) [inline src/map/map.h]: should not be 3rd class

File: db/constants.conf [hunk 5]
```diff
@@ -325,13 +332,15 @@ constants_db: {
...
 	EAJ_GUILLOTINE_CROSS:      0x4106
+	EAJ_STAR_EMPEROR:          0x4107
 
 	EAJ_ROYAL_GUARD:           0x4201
 	EAJ_SORCERER:              0x4202
 	EAJ_MINSTRELWANDERER:      0x4203
 	EAJ_SURA:                  0x4204
 	EAJ_GENETIC:               0x4205
 	EAJ_SHADOW_CHASER:         0x4206
+	EAJ_SOUL_REAPER:           0x4207
```
Comment (dastgirp) [inline db/constants.conf]: maybe just 0x1207?

File: db/constants.conf [hunk 6]
```diff
@@ -325,13 +332,15 @@ constants_db: {
 	EAJ_ARCH_BISHOP:           0x4104
 	EAJ_MECHANIC:              0x4105
 	EAJ_GUILLOTINE_CROSS:      0x4106
+	EAJ_STAR_EMPEROR:          0x4107
```
Comment (dastgirp) [inline db/constants.conf]: 0x1107? Same with others, because it is clearly mentioned it is not 3rd job

**Comment by Asheraf** [comment]
closing this for now as i don't feel like working on it, if someone would like to pick it up go ahead.

**Pull request closed without merging**